tests/specification_tests/test_client_specs.py

```diff
@@ -0,0 +1,81 @@
+import json
+import uuid
+from os import path
+from UnleashClient import UnleashClient
+from tests.utilities.testing_constants import URL, APP_NAME
+from UnleashClient.cache import FileCache
+import pytest
+
+
+CLIENT_SPEC_PATH = "tests/specification_tests/client-specification/specifications"
```

Comment:
The way we do it for the Java client is that we're explicitly targeting a version of the client-specification, so if new specifications are added that are not supported yet, they will fail if we try to target a newer client-specification until we've actually implemented it, but since you're locked to a specific version of the client-specifications, there shouldn't be a way a new case/spec can be added. From the GH workflow here, the python SDK does the same, it points to an specific version.

Reply:
What Chris said! The clone will pull from a tag in the repo so this is effectively pinning this to a tag in the client spec, which should never change. This does mean we'll have to remember to keep the pipeline up to date when we add new features but we're planning on handling monitoring feature completeness against the specs in a separate project